Public/Src/Cache/ContentStore/Library/FileSystem/TrackingFileStream.cs

```diff
@@ -0,0 +1,163 @@
+            : base(path, mode, access, share, bufferSize, options)
+        {
+            Path = path;
+        }
+
+        /// <inheritdoc />
+        public TrackingFileStream([NotNull] string path, FileMode mode, FileAccess access, FileShare share, int bufferSize, bool useAsync)
+            : base(path, mode, access, share, bufferSize, useAsync)
+        {
+            Path = path;
+        }
+
+        /// <inheritdoc />
+        public TrackingFileStream(string path, FileMode mode, FileSystemRights rights, FileShare share, int bufferSize, FileOptions options, FileSecurity fileSecurity)
+            : base(path, mode, rights, share, bufferSize, options, fileSecurity)
+        {
+        }
+
+        /// <inheritdoc />
+        public TrackingFileStream(string path, FileMode mode, FileSystemRights rights, FileShare share, int bufferSize, FileOptions options)
+            : base(path, mode, rights, share, bufferSize, options)
+        {
+            Path = path;
+        }
+
+        /// <inheritdoc />
+        [Obsolete]
+        public TrackingFileStream(IntPtr handle, FileAccess access)
+            : base(handle, access)
+        {
+        }
+
+        /// <inheritdoc />
+        [Obsolete]
+        public TrackingFileStream(IntPtr handle, FileAccess access, bool ownsHandle)
+            : base(handle, access, ownsHandle)
+        {
+        }
+
+        /// <inheritdoc />
+        [Obsolete]
+        public TrackingFileStream(IntPtr handle, FileAccess access, bool ownsHandle, int bufferSize)
+            : base(handle, access, ownsHandle, bufferSize)
+        {
+        }
+
+        /// <inheritdoc />
+        [Obsolete]
+        public TrackingFileStream(IntPtr handle, FileAccess access, bool ownsHandle, int bufferSize, bool isAsync)
+            : base(handle, access, ownsHandle, bufferSize, isAsync)
+        {
+        }
+
+        /// <inheritdoc />
+        public TrackingFileStream([NotNull] SafeFileHandle handle, FileAccess access, string path)
+            : base(handle, access)
+        {
+            Path = path;
+        }
+
+        /// <inheritdoc />
+        public TrackingFileStream([NotNull] SafeFileHandle handle, FileAccess access, int bufferSize, string path)
+            : base(handle, access, bufferSize)
+        {
+            Path = path;
+        }
+
+        /// <inheritdoc />
+        public TrackingFileStream([NotNull] SafeFileHandle handle, FileAccess access, int bufferSize, bool isAsync, string path)
+            : base(handle, access, bufferSize, isAsync)
+        {
+            Path = path;
+        }
+
+        /// <inheritdoc />
+        public override void Close()
+        {
+            Interlocked.Increment(ref ProperlyClosed);
+            GC.SuppressFinalize(this);
+        }
+
+        ~TrackingFileStream()
+        {
+            Interlocked.Increment(ref Leaked);
+
+            try

```

Comment:
Do we want to somehow log (maybe by leaving a file in a well-known-directory) that the finalizer ran at all?

Reply:
Never mind, I see you do this via statistics.